### PR TITLE
Added low-level configs examples

### PR DESCRIPTION
**Pull request by lsetiawan**

### CODE REVIEW

File: wof/examples/configs/timeseries/timeseries [hunk 1]
```diff
@@ -0,0 +1,9 @@
+server {
+    listen 8080;
+    server_name 127.0.0.1;
+
+    location /odm2timeseries {
```
Comment (lsetiawan) [inline wof/examples/configs/timeseries/timeseries]: If I change the application name to `wofpy` there will be a `wofpy.sock`. Assuming this is for 2 dao's not sure how to separate them within NGINX...

File: wof/examples/configs/timeseries/timeseries.conf [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# description of the script purpose
```
Comment (lsetiawan) [inline wof/examples/configs/timeseries/timeseries.conf]: Filename `timeseries.conf` is too general. I think best way is to make this `wofpy.conf` to server multiple dao also.

File: wof/examples/configs/timeseries/timeseries.conf [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+# The Nginx server runs under the www-data group.
+# We need Nginx to be able to read from and write to the socket file, so we'll give this group ownership over the process
+setuid www-data
+setgid www-data
+
+# Set the conda environment path
+env PATH=/home/ubuntu/miniconda/envs/wofpy/bin
+
+# change directory to where the uWSGI configuration file is located then execute uwsgi with the config file
+chdir /var/www/timeseries
```
Comment (lsetiawan) [inline wof/examples/configs/timeseries/timeseries.conf]: To make this more broad, change to `/var/www/wofpy` for a `wofpy` application

**Comment by lsetiawan** [comment]
@emiliom I've updated the `production_config` files. Let me know what you think. I've generalized things. Thanks.

**Comment by emiliom** [comment]
@lsetiawan, can you provide a summary of what you changed, what these config files are doing now that's different from before? I can see you've implemented many of the ideas we discussed last week (eg, new target directory names, new file names for these config files, etc), but I find it hard to follow which ones specifically. Thanks!

File: wof/examples/production_configs/wofpy [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    listen 8080;
+
+    # Can be assigned a domain name here
+    # Ex. example.com
+    # http://nginx.org/en/docs/http/server_names.html
+    server_name 127.0.0.1;
+
+    # Add whatever is necessary to cover all your url
+    # configured within the .cfg file
+    # Ex. Network: ODM2Timeseries this will result
```
Comment (lsetiawan) [inline wof/examples/production_configs/wofpy]: I provided some comments about the setup for `nginx` for example the location URL is based on configuration within the `*.cfg` file

File: wof/examples/production_configs/wofpy.conf [hunk 5]
```diff
@@ -1,5 +1,5 @@
 # description of the script purpose
-description "uWSGI server instance configured to serve timeseries"
+description "uWSGI server instance configured to serve WOFpy"
```
Comment (lsetiawan) [inline wof/examples/production_configs/wofpy.conf]: I've generalized the setup so that it would work for multiple dao's or wofpy app such as having `timeseries` and `measurement` etc.

File: wof/examples/production_configs/wsgi.py [hunk 6]
```diff
@@ -1,6 +1,3 @@
 #! /home/ubuntu/miniconda/envs/wofpy/bin/python
-import sys
-sys.path.insert(0, "/var/www/timeseries")
```
Comment (lsetiawan) [inline wof/examples/production_configs/wsgi.py]: This doesn't seem to be needed since these setting will be in the same folder as `runserver.py`

**Comment by lsetiawan** [comment]
In summary, the settings are not doing anything different per se, but now it's more generalized as a production setting for `wofpy` app rather than a unique dao such as `timeseries`

File: wof/examples/production_configs/wofpy (hunk 4 above)
Comment (emiliom) [inline wof/examples/production_configs/wofpy]: Ok

File: wof/examples/production_configs/wsgi.py (hunk 6 above)
Comment (emiliom) [inline wof/examples/production_configs/wsgi.py]: Great. Less hard wiring.

File: wof/examples/production_configs/wofpy.conf (hunk 5 above)
Comment (emiliom) [inline wof/examples/production_configs/wofpy.conf]: Can you point out where that generalization was done? Your in-line comment only refers to a description, and I don't see anything obvious in the rest of the file that points to that generalization.

File: wof/examples/production_configs/wofpy.conf [hunk 7]
```diff
@@ -15,5 +15,5 @@ setgid www-data
 env PATH=/home/ubuntu/miniconda/envs/wofpy/bin
 
 # change directory to where the uWSGI configuration file is located then execute uwsgi with the config file
-chdir /var/www/timeseries
-exec uwsgi --ini timeseries.ini
\ No newline at end of file
+chdir /var/www/wofpy
+exec uwsgi --ini wofpy.ini
```
Comment (lsetiawan) [inline wof/examples/production_configs/wofpy.conf]: > Can you point out where that generalization was done? Your in-line comment only refers to a description, and I don't see anything obvious in the rest of the file that points to that generalization.

Basically rather than `timeseries` it is now just called `wofpy` and when a user start or stop the server, they would only need to type for example `systemctl start wofpy` etc.

Comment (lsetiawan) [inline wof/examples/production_configs/wofpy.conf]: The actual implementation of multiple dao's would be in the `runserver` script. In this case you'd use the `runserver_multiple.py`

Comment (emiliom) [inline wof/examples/production_configs/wofpy.conf]: Got it! So this means that a user who wants to run **more than one** service would still only run that single line `systemctl start wofpy`?!

Crossing fingers that the answer is yes :smiley: ... I don't see anything in `wofpy.conf`, `wofpy.ini` or `wofpy.service` that suggests they're limited to one wofpy service.

Comment (emiliom) [inline wof/examples/production_configs/wofpy.conf]: > The actual implementation of multiple dao's would be in the runserver script. In this case you'd use the runserver_multiple.py

I missed this when I made my comments ... 

It looks like the file `wofpy` would also need to be modified for a multi-service set up, no?

Comment (lsetiawan) [inline wof/examples/production_configs/wofpy.conf]: > Got it! So this means that a user who wants to run more than one service would still only run that single line systemctl start wofpy?!

Yes

> It looks like the file wofpy would also need to be modified for a multi-service set up, no?

Yep. Hence the comment in [L13](https://github.com/lsetiawan/WOFpy/blob/533f6d3bffdf26b987da7c2e8b3a45c0107759cf/wof/examples/production_configs/wofpy#L13) Just gotta add the URL Configurations for multiple dao's

Comment (emiliom) [inline wof/examples/production_configs/wofpy.conf]: Perfect! Sorry I took so long to review this, and had so many question/misunderstandings.

This is awesome. Merging now.
